cmd/server/controllers/sellers.go

```diff
@@ -0,0 +1,234 @@
+package controllers
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/marcoglnd/mercado-fresco-packmain/internal/sellers"
+)
+
+// Controller recebe um serviço
+type SellerController struct {
+	service sellers.Service
+}
+
+func NewSeller(s sellers.Service) *SellerController {
+	return &SellerController{
+		service: s,
+	}
+}
+
+// @Summary List sellers
+// @Tags Sellers
+// @Description get all sellers
+// @Accept json
+// @Produce json
+// @Param token header string true "token"
+// @Success 200 {object} schemes.JSONSuccessResult{data=schemes.Seller}
+// @Failure 404 {object} schemes.JSONBadReqResult{error=string}
+// @Router /sellers [get]
+func (c *SellerController) GetAll() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		token := ctx.Request.Header.Get("token")
+		if token != "123456" {
+			ctx.JSON(401, gin.H{
+				"error": "token inválido",
+			})
+			return
+		}
+
+		s, err := c.service.GetAll()
+		if err != nil {
+			ctx.JSON(404, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+
+		ctx.JSON(200, s)
+	}
+}
+
+// @Summary Seller by id
+// @Tags Sellers
+// @Description get Seller by it's id
+// @Accept json
+// @Produce json
+// @Param id path int true "Seller ID"
+// @Param token header string true "token"
+// @Success 200 {object} schemes.JSONSuccessResult{data=schemes.Seller}
+// @Failure 400 {object} schemes.JSONBadReqResult{error=string}
+// @Failure 404 {object} schemes.JSONBadReqResult{error=string}
+// @Router /sellers/{id} [get]
+func (c *SellerController) GetById() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		id := ctx.Param("id")
+		intId, err := strconv.Atoi(id)
+		if err != nil {
+			ctx.JSON(http.StatusBadRequest, gin.H{"error": "Invalid id"})
+			return
+		}
+		s, err := c.service.GetById(intId)
+		if err != nil {
+			ctx.JSON(http.StatusNotFound, gin.H{"error": err.Error()})
+			return
+		}
+		ctx.JSON(http.StatusOK, s)
+	}
+}
+
+type requestSellers struct {
+	Cid          int    `json:"cid"`
+	Company_name string `json:"company_name"`
+	Address      string `json:"address"`
+	Telephone    int    `json:"telephone"`
+}
+
+// @Summary Create seller
+// @Tags Sellers
+// @Description Add a new Seller to the list
+// @Accept json
+// @Produce json
+// @Param token header string true "token"
+// @Param Seller body requestSellers true "seller to create"
+// @Success 201 {object} schemes.JSONSuccessResult{data=schemes.Seller}
+// @Failure 404 {object} schemes.JSONBadReqResult{error=string}
+// @Failure 422 {object} schemes.JSONBadReqResult{error=string}
+// @Router /sellers [post]
+func (c *SellerController) CreateNewSeller() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		token := ctx.Request.Header.Get("token")
+		if token != "123456" {
+			ctx.JSON(http.StatusUnauthorized, gin.H{"error": "token inválido"})
+			return
+		}
+
+		var req requestSellers
+		if err := ctx.Bind(&req); err != nil {
+			ctx.JSON(http.StatusNotFound, gin.H{
```

Comment:
O que acha de trocar esse `StatusNotFound` por `StatusUnprocessableEntity`?
Como estamos verificando se os dados condizem com a struct `requestSellers` melhoraria na legibilidade do erro
```suggestion
			ctx.JSON(http.StatusUnprocessableEntity, gin.H{
```